operator/src/main/java/oracle/kubernetes/operator/helpers/DomainValidationSteps.java

```diff
@@ -267,8 +267,23 @@ private List<ClusterResource> getClusters(Packet packet) {
 
     @Override
     public ClusterResource findCluster(V1LocalObjectReference reference) {
-      return Optional.ofNullable(reference.getName()).flatMap(name -> getClusters(packet).stream()
-          .filter(cluster -> name.equals(cluster.getClusterName())).findFirst()).orElse(null);
+      return Optional.ofNullable(reference.getName())
+          .flatMap(name -> Optional.ofNullable(getClusters(packet))
+              .orElse(Collections.emptyList())
+              .stream()
+              .filter(cluster -> name.equals(cluster.getMetadata().getName()))
+              .findFirst())
+          .orElse(null);
+    }
+
+    @Override
+    public ClusterResource findClusterInNamespace(V1LocalObjectReference reference, String namespace) {
+      return Optional.ofNullable(reference.getName())
+          .flatMap(name -> Optional.of(DomainProcessorImpl.getClusters(namespace))
```

Comment:
Do we need to use Optional here since DPI.getClusters() always return an ArrayList?

Reply:
No, we don't need to use Optional here. Will Change. thanks